drivers/flash/soc_flash_mcux.c

```diff
@@ -85,13 +85,24 @@ static int flash_mcux_write_protection(struct device *dev, bool enable)
 	return -EIO;
 }
 
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+static void flash_page_layout_not_implemented(void)
+{
+	k_panic();
+}
+#endif
+
```

Comment:
Rather than duplicating this in all the drivers, why can't we do this at the caller of api->page_layout?  It can check for null to decide to assert or not?

Reply:
This will add additional latency and code. I'm always concerned on this for low level embedded driver. As each driver is independent a entity I see sense in duplication of code here. I expect in near feature this features will be implemented for most of te driver, even so NFFS requires this extension or mcuboot recommends this extension as well.

Reply:
Can we at least put the implementation of flash_page_layout_not_implemented in a common header so each driver isn't implementing it differently?  (header should be in drivers/flash/flash_priv.h or something like that)